src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.cs

```diff
@@ -44,9 +44,12 @@ private enum TimeZoneInfoResult
         private const int MaxKeyLength = 255;
 
         private readonly string _id;
-        private readonly string? _displayName;
-        private readonly string? _standardDisplayName;
-        private readonly string? _daylightDisplayName;
+        private bool _displayNameSet;
```

Comment:
>_displayNameSet [](http://example.com/codeflow?start=21&length=15)

Instead of introducing these variables, would it make sense when creating custom time zone with null names, convert the names to empty strings at that time? we always return empty string anyway when we have null names.

Reply:
Good point, that way we can still honor that the user intentionally set them to `null`. Removed the boolean variables and added `?? string.Empty` in the `CreateCustomTimeZone` APIs